pkg/commands/workdir.go

```diff
@@ -81,14 +84,116 @@ func (w *WorkdirCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile
 
 // FilesToSnapshot returns the workingdir, which should have been created if it didn't already exist
 func (w *WorkdirCommand) FilesToSnapshot() []string {
-	return w.snapshotFiles
+	return nil
+}
+
+func (r *WorkdirCommand) ProvidesFilesToSnapshot() bool {
+	return false
 }
 
 // String returns some information about the command for the image config history
 func (w *WorkdirCommand) String() string {
 	return w.cmd.String()
 }
 
+// CacheCommand returns true since this command should be cached
+func (w *WorkdirCommand) CacheCommand(img v1.Image) DockerCommand {
+
+	return &CachingWorkdirCommand{
+		img:       img,
+		cmd:       w.cmd,
+		extractFn: util.ExtractFile,
+	}
+}
+
 func (w *WorkdirCommand) MetadataOnly() bool {
 	return false
 }
+
+func (r *WorkdirCommand) RequiresUnpackedFS() bool {
+	return true
+}
+
+func (w *WorkdirCommand) ShouldCacheOutput() bool {
+	return w.shdCache
+}
+
+type CachingWorkdirCommand struct {
+	BaseCommand
+	caching
+	img            v1.Image
+	extractedFiles []string
+	cmd            *instructions.WorkdirCommand
+	extractFn      util.ExtractFunction
+}
+
+func (wr *CachingWorkdirCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.BuildArgs) error {
+	var err error
+	logrus.Info("Cmd: workdir")
```

Comment:
I copied that entire block of code here, it is to ensure that even if we hit the cache we still do the metadata operation thingy and actually change the workdir. I could of course put that in a function etc. but I would actually prefer to pass the resolved directory into the cache, s.t. I can have a single line here. Not only reusing the functionality, but reusing the result.

```go
cfg.WorkingDir = wr.resolvedWorkingDir
```

Reply:
passing the result is not possible, as either the regular or the cached ExecuteCommand function is called, never both.

Reply:
hmmm... passing the result would be possible but requires some creative thinking. We do store the layer as an image when we push to the cache, so we could change the WORKDIR on that image and with that store that variable too. Come to think of it, labels on the cached image could be used to pass arbitrary data between cache creation and reusing stage.